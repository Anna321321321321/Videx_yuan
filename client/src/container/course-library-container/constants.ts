export const NAME = 'courses-library';
