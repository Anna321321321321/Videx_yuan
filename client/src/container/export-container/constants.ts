export enum ExportStatus {
  Init,
  Processing,
  Success,
  Error
}
