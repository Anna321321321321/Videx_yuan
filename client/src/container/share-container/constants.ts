export enum ShareStatus {
  Init,
  Processing,
  Success,
  Error,
}
