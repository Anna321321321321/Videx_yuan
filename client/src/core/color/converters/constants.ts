import { TColorHex, TColorRbg, TColorString } from '../types';

export const color2hex = {
  red: '#e32990' as TColorHex,
  yellow: '#fff110' as TColorHex,
  green: '#4cba35' as TColorHex,
  blue: '#28a3dc' as TColorHex,
  purple: '#9719ff' as TColorHex
};

export const hex2color = {
  '#e32990': 'red' as TColorString,
  '#28a3dc': 'blue' as TColorString,
  '#9719ff': 'purple' as TColorString,
  '#4cba35': 'green' as TColorString,
  '#fff110': 'yellow' as TColorString,
  '#ffffff': 'white' as TColorString
};

export const hex2rgb = {
  '#e32990': 'rgba(227,41,144,0.2)' as TColorRbg,
  '#28a3dc': 'rgba(40,163,220,0.2)' as TColorRbg,
  '#9719ff': 'rgba(151,25,255,0.2)' as TColorRbg,
  '#4cba35': 'rgba(76,186,53,0.2)' as TColorRbg,
  '#fff110': 'rgba(255,241,16,0.2)' as TColorRbg
};

export const hex2rgbOpaque = {
  '#e32990': 'rgba(227,41,144,0.8)' as TColorRbg,
  '#28a3dc': 'rgba(40,163,220,0.8)' as TColorRbg,
  '#9719ff': 'rgba(151,25,255,0.8)' as TColorRbg,
  '#4cba35': 'rgba(76,186,53,0.8)' as TColorRbg,
  '#fff110': 'rgba(255,241,16,0.8)' as TColorRbg
};
