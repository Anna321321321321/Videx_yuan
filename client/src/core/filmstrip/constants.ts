/**
 * Filmstrip HEIGHT
 *
 * @type {number}
 */
export const FILMSTRIP_HEIGHT: number = 150;

export const HIGHLIGHT_COLOR_COUNT: number = 4;
