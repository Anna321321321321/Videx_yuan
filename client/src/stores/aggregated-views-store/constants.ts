export const NAME = 'aggregated-views-store';
