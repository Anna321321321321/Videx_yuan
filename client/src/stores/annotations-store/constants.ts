export const NAME = 'annotations-store';
