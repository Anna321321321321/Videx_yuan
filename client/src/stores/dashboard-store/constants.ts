export const NAME = 'dashboard-course';
