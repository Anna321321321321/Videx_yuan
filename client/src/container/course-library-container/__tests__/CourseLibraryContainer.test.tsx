import { mount, shallow } from 'enzyme';
import { fromJS } from 'immutable';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ConnectedCourseLibraryContainer, {
  CourseLibraryContainer
} from '../CourseLibraryContainer';
import * as constants from '../constants';

const unInitializedProps = {
  courses: [],
  metadata: {
    adminAccess: false
  },
  initialized: false
};

const initializedProps = {
  courses: [
    {
      name: 'test1',
      metadata: {
        token: 'test',
        ownerAccess: true
      },
      id: '1'
    }
  ],
  metadata: {
    adminAccess: true
  },
  initialized: true
};

describe('Course Library Container Unit Tests', () => {
  it('Mount & Unmount component', () => {
    const mockInit = jest.fn();
    const mockDeinit = jest.fn();
    const wrapper = shallow(
      <CourseLibraryContainer
        {...unInitializedProps}
        init={mockInit}
        deinit={mockDeinit}
      />
    );
    expect(mockInit.mock.calls.length).toBe(1);
    wrapper.unmount();
    expect(mockDeinit.mock.calls.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render uninitialized component', () => {
    const mockInit = jest.fn();
    const mockDeinit = jest.fn();
    const mockDeleteCourse = jest.fn();
    const wrapper = shallow(
      <CourseLibraryContainer
        {...unInitializedProps}
        init={mockInit}
        deinit={mockDeinit}
      />
    );
    expect(wrapper.find('Spin').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render initialized component', () => {
    const mockInit = jest.fn();
    const mockDeinit = jest.fn();
    const mockDeleteCourse = jest.fn();
    const wrapper = shallow(
      <CourseLibraryContainer
        {...initializedProps}
        init={mockInit}
        deinit={mockDeinit}
      />
    );
    expect(wrapper.find('Spin').length).toBe(0);
    expect(wrapper.find('CourseSider').length).toBe(1);
    expect(wrapper.find('CourseCard').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Connected CourseLibrary Container Unit Tests with Redux Store', () => {
  const initialState = {
    [constants.NAME]: fromJS(unInitializedProps)
  };
  const mockStore = configureStore();

  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedCourseLibraryContainer store={store} />);
  });

  it('Render connected component', () => {
    expect(container.length).toEqual(1);
    expect(container).toMatchSnapshot();
  });

  it('Check props matches with init state', () => {
    expect(container.prop('courses')).toEqual([]);
    expect(container.prop('metadata')).toEqual({ adminAccess: false });
    expect(container.prop('initialized')).toEqual(false);
    expect(container).toMatchSnapshot();
  });
});

describe('Connected CourseLibrary Container Unit Tests Full Render with Redux Store', () => {
  const initialState = {
    [constants.NAME]: fromJS(unInitializedProps)
  };
  const mockStore = configureStore();

  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    CourseLibraryContainer.prototype.componentDidMount = null;
    CourseLibraryContainer.prototype.componentWillUnmount = null;
    wrapper = mount(
      <Provider store={store}>
        <ConnectedCourseLibraryContainer />
      </Provider>
    );
  });

  it('Render connected component', () => {
    expect(wrapper.find('CourseLibraryContainer').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('Check props matches with init state', () => {
    const component = wrapper.find('CourseLibraryContainer');
    expect(component.prop('courses')).toEqual([]);
    expect(component.prop('metadata')).toEqual({ adminAccess: false });
    expect(component.prop('initialized')).toEqual(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('Check INIT action on dispatching', () => {
    let action;
    store.dispatch({
      type: 'courses-library/INIT',
      payload: {
        courses: [
          {
            name: 'test1',
            metadata: {
              contributorAccess: true,
              ownerAccess: true
            },
            id: '1'
          }
        ],
        metadata: {
          adminAccess: true
        }
      }
    });
    action = store.getActions();
    expect(action[0].type).toBe('courses-library/INIT');
    expect(wrapper).toMatchSnapshot();
  });

  it('Check DEINIT action on dispatching', () => {
    let action;
    store.dispatch({
      type: 'courses-library/DEINIT'
    });
    action = store.getActions();
    expect(action[0].type).toBe('courses-library/DEINIT');
    expect(wrapper).toMatchSnapshot();
  });
});
