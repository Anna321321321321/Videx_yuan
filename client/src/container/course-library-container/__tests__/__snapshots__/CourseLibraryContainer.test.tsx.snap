// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected CourseLibrary Container Unit Tests Full Render with Redux Store Check DEINIT action on dispatching 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CourseLibraryContainer)>
    <CourseLibraryContainer
      courses={Array []}
      deinit={[Function]}
      init={[Function]}
      initialized={false}
      metadata={
        Object {
          "adminAccess": false,
        }
      }
    >
      <Adapter
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <BasicLayout
          prefixCls="ant-layout"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="ant-layout"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <Spin
              prefixCls="ant-spin"
              size="large"
              spinning={true}
              wrapperClassName=""
            >
              <div
                className="ant-spin ant-spin-lg ant-spin-spinning"
              >
                <span
                  className="ant-spin-dot ant-spin-dot-spin"
                >
                  <i />
                  <i />
                  <i />
                  <i />
                </span>
              </div>
            </Spin>
          </div>
        </BasicLayout>
      </Adapter>
    </CourseLibraryContainer>
  </Connect(CourseLibraryContainer)>
</Provider>
`;

exports[`Connected CourseLibrary Container Unit Tests Full Render with Redux Store Check INIT action on dispatching 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CourseLibraryContainer)>
    <CourseLibraryContainer
      courses={Array []}
      deinit={[Function]}
      init={[Function]}
      initialized={false}
      metadata={
        Object {
          "adminAccess": false,
        }
      }
    >
      <Adapter
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <BasicLayout
          prefixCls="ant-layout"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="ant-layout"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <Spin
              prefixCls="ant-spin"
              size="large"
              spinning={true}
              wrapperClassName=""
            >
              <div
                className="ant-spin ant-spin-lg ant-spin-spinning"
              >
                <span
                  className="ant-spin-dot ant-spin-dot-spin"
                >
                  <i />
                  <i />
                  <i />
                  <i />
                </span>
              </div>
            </Spin>
          </div>
        </BasicLayout>
      </Adapter>
    </CourseLibraryContainer>
  </Connect(CourseLibraryContainer)>
</Provider>
`;

exports[`Connected CourseLibrary Container Unit Tests Full Render with Redux Store Check props matches with init state 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CourseLibraryContainer)>
    <CourseLibraryContainer
      courses={Array []}
      deinit={[Function]}
      init={[Function]}
      initialized={false}
      metadata={
        Object {
          "adminAccess": false,
        }
      }
    >
      <Adapter
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <BasicLayout
          prefixCls="ant-layout"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="ant-layout"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <Spin
              prefixCls="ant-spin"
              size="large"
              spinning={true}
              wrapperClassName=""
            >
              <div
                className="ant-spin ant-spin-lg ant-spin-spinning"
              >
                <span
                  className="ant-spin-dot ant-spin-dot-spin"
                >
                  <i />
                  <i />
                  <i />
                  <i />
                </span>
              </div>
            </Spin>
          </div>
        </BasicLayout>
      </Adapter>
    </CourseLibraryContainer>
  </Connect(CourseLibraryContainer)>
</Provider>
`;

exports[`Connected CourseLibrary Container Unit Tests Full Render with Redux Store Render connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CourseLibraryContainer)>
    <CourseLibraryContainer
      courses={Array []}
      deinit={[Function]}
      init={[Function]}
      initialized={false}
      metadata={
        Object {
          "adminAccess": false,
        }
      }
    >
      <Adapter
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <BasicLayout
          prefixCls="ant-layout"
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="ant-layout"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
              }
            }
          >
            <Spin
              prefixCls="ant-spin"
              size="large"
              spinning={true}
              wrapperClassName=""
            >
              <div
                className="ant-spin ant-spin-lg ant-spin-spinning"
              >
                <span
                  className="ant-spin-dot ant-spin-dot-spin"
                >
                  <i />
                  <i />
                  <i />
                  <i />
                </span>
              </div>
            </Spin>
          </div>
        </BasicLayout>
      </Adapter>
    </CourseLibraryContainer>
  </Connect(CourseLibraryContainer)>
</Provider>
`;

exports[`Connected CourseLibrary Container Unit Tests with Redux Store Check props matches with init state 1`] = `
<CourseLibraryContainer
  courses={Array []}
  deinit={[Function]}
  init={[Function]}
  initialized={false}
  metadata={
    Object {
      "adminAccess": false,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`Connected CourseLibrary Container Unit Tests with Redux Store Render connected component 1`] = `
<CourseLibraryContainer
  courses={Array []}
  deinit={[Function]}
  init={[Function]}
  initialized={false}
  metadata={
    Object {
      "adminAccess": false,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`Course Library Container Unit Tests Mount & Unmount component 1`] = `
<Adapter
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <Spin
    prefixCls="ant-spin"
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</Adapter>
`;

exports[`Course Library Container Unit Tests Render initialized component 1`] = `
<Adapter
  className="ant-layout-has-sider"
>
  <CourseSider
    adminAccess={true}
    courses={
      Array [
        Object {
          "id": "1",
          "metadata": Object {
            "ownerAccess": true,
            "token": "test",
          },
          "name": "test1",
        },
      ]
    }
  />
  <Adapter
    style={
      Object {
        "padding": "0 24px",
      }
    }
  >
    <Adapter
      style={
        Object {
          "margin": 0,
          "minHeight": 280,
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
        }
      }
    >
      <CourseCard
        id="1"
        key="0"
        metadata={
          Object {
            "ownerAccess": true,
            "token": "test",
          }
        }
        name="test1"
        onDeleteCallback={[Function]}
      />
    </Adapter>
  </Adapter>
</Adapter>
`;

exports[`Course Library Container Unit Tests Render uninitialized component 1`] = `
<Adapter
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <Spin
    prefixCls="ant-spin"
    size="large"
    spinning={true}
    wrapperClassName=""
  />
</Adapter>
`;
